native_client/deepspeech.h

```diff
@@ -109,6 +122,44 @@ char* DS_SpeechToText(ModelState* aCtx,
                       unsigned int aBufferSize,
                       unsigned int aSampleRate);
 
+/**
+ * @brief Use the DeepSpeech model to perform Speech-To-Text and output metadata 
+ * about the results.
+ *
+ * @param aCtx The ModelState pointer for the model to use.
+ * @param aBuffer A 16-bit, mono raw audio signal at the appropriate
+ *                sample rate.
+ * @param aBufferSize The number of samples in the audio signal.
+ * @param aSampleRate The sample-rate of the audio signal.
+ *
+ * @return Outputs a struct of individual letters along with their timing information. 
+ * 		   The user is responsible for freeing Metadata and Metadata.items. Returns NULL on error.
+ */
+DEEPSPEECH_EXPORT
+Metadata* DS_SpeechToTextWithMetadata(ModelState* aCtx,
+                      const short* aBuffer,
+                      unsigned int aBufferSize,
+                      unsigned int aSampleRate);
+
+/**
+ * @brief Sets up the streaming state and feeds audio samples to it. {@link DS_FinishStream()} or 
+ * {@link DS_FinishStreamWithMetadata()} can then be called on the StreamingState object to 
+ * get inference results.
+ *
+ * @param aCtx The ModelState pointer for the model to use.
+ * @param aBuffer A 16-bit, mono raw audio signal at the appropriate
+ *                sample rate.
+ * @param aBufferSize The number of samples in the audio signal.
+ * @param aSampleRate The sample-rate of the audio signal.
+ *
+ * @return Outputs a StreamingState object to pass to {@link DS_FinishStream()} or 
+ * {@link DS_FinishStreamWithMetadata()}. Returns NULL on error.
+ */
+DEEPSPEECH_EXPORT
+StreamingState* DS_SetupStreamAndFeedAudioContent(ModelState* aCtx,
```

Comment:
I don't think this should be exposed, just a private function in deepspeech.cc is enough, as this is just a simple composition of other API functions.